### PR TITLE
gh-98284: better error message for undefined abstractmethod

### PR DESCRIPTION
**Pull request by kaushikcfd**
Error message before:
```
TypeError: Can't instantiate abstract class IndexNode with abstract method name
```

Error message after this patch:
```
TypeError: Can't instantiate abstract class IndexNode with abstract method 'name'
```

Given that the style-guide requires method names to be in lowercase, the quotation marks help in differentiating the sentence from the method's name.

* Issue: gh-98284

### CODE REVIEW

File: Lib/test/test_abc.py [hunk 1]
```diff
@@ -165,7 +165,7 @@ def method_one(self):
                 @abc.abstractmethod
                 def method_two(self):
                     pass
-            msg = r"class C without an implementation for abstract methods method_one, method_two"
+            msg = r"class C without an implementation for abstract methods 'method_one, method_two'"
```
Comment (JelleZijlstra) [inline Lib/test/test_abc.py]: This is incorrect, we should quote each method individually.

Comment (kaushikcfd) [reply Lib/test/test_abc.py]: Thanks, done!

**Comment by JelleZijlstra** [comment]
You do need a NEWS entry as the good knight Bedevere keeps telling you.

Also, avoid force-pushing; it makes review harder. We'll squash-merge anywhere when we merge a PR.

**Comment by kaushikcfd** [comment]
> You do need a NEWS entry as the good knight Bedevere keeps telling you.

Done!

> Also, avoid force-pushing;

Noted, thanks!

File: Misc/NEWS.d/next/Library/2022-10-09-18-08-08.gh-issue-97971.SaVHTd.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Improved :class:`TypeError` message for undefined abstract methods of a
```
Comment (JelleZijlstra) [inline Misc/NEWS.d/next/Library/2022-10-09-18-08-08.gh-issue-97971.SaVHTd.rst]: The NEWS entry must refer to an issue, not a PR. You'll have to create an issue and rename the file to have your issue number in the file name.

Comment (kaushikcfd) [reply Misc/NEWS.d/next/Library/2022-10-09-18-08-08.gh-issue-97971.SaVHTd.rst]: Thanks, see [4073b1f](https://github.com/python/cpython/pull/97971/commits/4073b1fb4a0afe64c5c5742c75b12d80262bae31).

**Comment by kaushikcfd** [comment]
The CI failure is (most likely) unrelated:
```
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'D:\\a\\cpython\\cpython\\build\\test_python_2628�\\test_python_worker_1152�'
```

**Comment by sobolevn** [comment]
Retriggering CI, it is indeed unrelated.

**Comment by kaushikcfd** [comment]
> Retriggering CI, it is indeed unrelated.

Thanks, yep that was it.

**Comment by kaushikcfd** [comment]
Soft ping.

File: Objects/typeobject.c [hunk 3]
```diff
@@ -4887,11 +4888,12 @@ object_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
 
         PyErr_Format(PyExc_TypeError,
                      "Can't instantiate abstract class %s "
-                     "without an implementation for abstract method%s %U",
+                     "without an implementation for abstract method%s '%U'",
                      type->tp_name,
                      method_count > 1 ? "s" : "",
                      joined);
         Py_DECREF(joined);
+        Py_DECREF(comma_w_quotes_sep);
```
Comment (JelleZijlstra) [inline Objects/typeobject.c:4901]: You also need to DECREF this in the two `return NULL` branches above.

Come to think of it, the second of those already leaks `joined`. It needs a `Py_DECREF(joined)`.

**Review by JelleZijlstra — APPROVED** [review]

File: Objects/typeobject.c (hunk 3 above)
Comment (kaushikcfd) [reply Objects/typeobject.c:4901]: Aah, yep! Fixed in [6ae052e](https://github.com/python/cpython/pull/97971/commits/6ae052ef771eb9e72f7cbb90b58a2bb1d10ae3f5).

**Pull request merged**